core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -550,4 +598,72 @@ static void prepareShrinkIndexSettings(ClusterState currentState, Set<String> ma
             .put(IndexMetaData.INDEX_SHRINK_SOURCE_UUID.getKey(), shrinkFromIndex.getUUID());
     }
 
+    private void waitOnShards(final String indexName,
+                              final ActiveShardCount waitForActiveShards,
+                              final ActionListener<ClusterStateUpdateResponse> listener,
+                              final TimeValue timeout,
+                              final ThreadContext threadContext) {
```

Comment:
no need to pass ThreadContext here. Just do `threadPool.getThreadContext()` when you create the observer.